vsphere_server_plugin/server.py

```diff
@@ -385,8 +387,8 @@ def create(server_client,
             raise NonRecoverableError(
                 'A VM with name {0} was not found.'.format(server.get('name')))
         ctx.instance.runtime_properties[VSPHERE_RESOURCE_EXTERNAL] = True
-    elif "template" not in server:
-        raise NonRecoverableError('No template provided.')
+    elif "template" not in server and "clone_vm" not in server:
```

Comment:
The original design of the plugin used double quotes. We have had a few re-factorings where we try to correct this. Please don't add more double quotes. 
```suggestion
    elif 'template' not in server and 'clone_vm' not in server:
```